docs-src/concepts.md

```diff
@@ -112,42 +114,123 @@ TODO
+  controller: "acme.io/gateway-controller"
 ```
 
-A sample controller file:
+We expect that one or more `GatewayClasses` will be created by the infrastructure
+provider for the user. It allows decoupling of which mechanism (e.g. controller)
+implements the `Gateways` from the user. For instance, an infrastructure
+provider may create two `GatewayClasses` named `internet`, `private` to reflect
+`Gateways` that define Internet-facing vs private, internal applications.
 
 ```yaml
-apiVersion: networking.acme.io/v1alpha1
-kind: CloudLB
+kind: GatewayClass
+metadata:
+  name: internet
+  ...
+---
+kind: GatewayClass
 metadata:
-  name: from-internet
-networkTier: auto
-ipAllocation: auto
+  name: private
+  ...
 ```
 
-We expect that one or more GatewayClasses will be created by the infrastructure
-provider for the user as part of their cloud infrastructure. If the cluster
-operator wishes to customize their `GatewayClasses`, they are free to do so,
-interacting with the custom resource parameters. Example parameters
-`GatewayClass.controller` resource will include cluster-level defaults and
-constraints.
+The user of the classes will not need to know *how* `internet` and `private` are
+implemented, whether they are cluster-based or cloud-based LBs.
 
 > The community may converge around a common set of GatewayClass names with
 > well-known meanings (e.g. `internet`, `private`) to ease interoperability
 > between platforms.
 
+#### GatewayClass parameters
+
+Providers of the `Gateway API` may need to pass parameters to their controller
+as part of the class definition. This done using the
+`GatewayClass.spec.parametersRef` field:
+
+```yaml
+# GatewayClass for Gateways that define Internet-facing applications.
+kind: GatewayClass
+metadata:
+  name: internet
+spec:
+  controller: "acme.io/gateway-controller"
+  parametersRef:
+    apiVersion: core/v1
+    kind: ConfigMap
+    namespace: acme-system
+    name: internet-gateway
+---
+kind: ConfigMap
+metadata:
+  name: internet-gateway
+  namespace: acme-system
+data:
+  ip-address-pool: internet-vips
+  ...
+```
+
+The type of object referenced by `GatewayClass.spec.parametersRef` will depend on
+the provider itself. A `core.ConfigMap` is used in the example above, but some
+controllers may opt to use a `CustomResource` for better schema validation.
+
+#### GatewayClass status
+
+`GatewayClasses` need to be validated by the provider to ensure that the
+configured parameters are valid. The validity of the class will be signaled to
+the user via `GatewayClass.status`:
+
+```yaml
+kind: GatewayClass
+...
+status:
+  conditions:
+  - type: InvalidParameters
+    ...
+```
+
+A new `GatewayClass` will start with the `InvalidParameters` condition set to
+`Unknown`. At this point the controller has not seen the configuration. Once the
+controller has processed the configuration, the condition will be set to
+`False`.
+
+```yaml
+kind: GatewayClass
+...
+status:
+  conditions:
+  - type: InvalidParameters
```

Comment:
Update `InvalidParameters` if https://github.com/kubernetes-sigs/service-apis/pull/45/files#r370279364 is accepted. The example would look like:
```
kind: GatewayClass
...
status:
  conditions:
  - type: Available
    status: False
    reason: InvalidParameters
    message: GatewayClass parameter "foo" is invalid.
    ...
```